### ISSUE TITLE
Authorized queries run and crash in server rendering

### ISSUE THREAD

**Issue by tmcw** [blitz-js/blitz#2964]
### What is the problem?

Routes that use `getServerProps` are rendered with SSR. If they use `useQuery`, the value of the query result, if you have a `resolver.authorize` call, is `undefined`. As a result, any page that relies on a `useQuery` value that is non-nullable and runs on the server will cause that route to crash in production.

Any of the following would be better:

- If `useQuery`'s result is nullable even if the query it's referring to is non-nullable, then its return type should be nullable. Right now, TypeScript allows you to rely on non-nullable query results that become nullable.
- If `useQuery` produced the correct result by actually running the authenticated query.
- If components with `useQuery` did not run.

### Paste all your error logs here:

```
This should not happen TypeError: Cannot read property 'email' of undefined
    at UserInfo (/Users/tmcw/p/sentry-example/.next/server/pages/index.js:137:29)
    at tb (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:55:138)
    at wb (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:57:254)
    at W (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:62:89)
    at X (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:64:97)
    at wb (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:59:313)
    at W (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:62:89)
    at X (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:64:97)
    at wb (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:58:27)
    at W (/Users/tmcw/p/sentry-example/node_modules/react-dom/cjs/react-dom-server-legacy.node.production.min.js:62:89)
```

### Paste all relevant code snippets here:

Example query resolver:

```ts
import { resolver, Ctx } from "blitz"
import db from "db"

export default resolver.pipe(
  resolver.authorize(),
  async function getCurrentUser(_ = null, { session }) {
    const user = await db.user.findFirst({
      where: { id: session.userId },
      select: { id: true, name: true, email: true, role: true },
      rejectOnNotFound: true,
    })

    return user
  }
)
```

Example from a slightly-modified new Blitz app that uses a query:

```tsx
const UserInfo = () => {
  const currentUser = useCurrentUser()
  try {
    console.log(currentUser.email)
  } catch (e) {
    console.log("This should not happen", e)
  }
  return null;
}
```

### What are detailed steps to reproduce this?

1. Have an app that uses `getServerSideProps` and an authenticated, non-nullquery

### Run `blitz -v` and paste the output here:

```
% blitz -v
macOS Monterey | darwin-x64 | Node: v16.7.0

blitz: 0.42.4 (global)
blitz: 0.42.4 (local)

  Package manager: yarn
  System:
    OS: macOS 12.0.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 2.93 GB / 16.00 GB
    Shell: 5.8 - /usr/local/bin/zsh
  Binaries:
    Node: 16.7.0 - ~/n/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.14.15 - /usr/local/bin/npm
    Watchman: Not Found
  npmPackages:
    @prisma/client: 3.4.2 => 3.4.2
    blitz: 0.42.4 => 0.42.4
    prisma: 3.4.2 => 3.4.2
    react: 18.0.0-alpha-5ca4b0433-20211020 => 18.0.0-alpha-5ca4b0433-20211020
    react-dom: 18.0.0-alpha-5ca4b0433-20211020 => 18.0.0-alpha-5ca4b0433-20211020
    typescript: ~4.4 => 4.4.4
```

**Comment by flybayer** [comment]
@tmcw unfortunately @beerose and I are unable to reproduce the issue in a new blitz app. Maybe there is a cache issue from your blitz upgrade? Try deleting `node_modules`, `.blitz`, and `.next`.

If that doesn't work, can you provide a full repo that we can clone and test? We'd love to get this fixed for you.

**Comment by tmcw** [comment]
Sorry, working on it. This bug is unbelievably hard to diagnose.

**Comment by tmcw** [comment]
Okay! I've finally distilled a repro!

https://github.com/tmcw/bug-distill

If you run

```
yarn
yarn build
yarn start
```

And request the first page, you should get the error.

**Comment by beerose** [comment]
Thanks for the repo! I can confirm that something weird happens and that the behaviour was different with Blitz 0.38. We'll investigate and get back to you as soon as possible

**Comment by beerose** [comment]
So in order to fix it, I first tried to create a minimal repro from a new Blitz app. I used @tmcw's code, but it was working fine, so I compared deps, and turns out it fails only with the experimental react version (new Blitz apps have 18 alpha so that works, and that's why we weren't able to reproduce it before). 

With experimental version and  `blitz start` runtime  `__BLITZ_SUSPENSE_ENABLED` is false in all the places, but I'm not sure what is causing it.
